benchmark/benchmark.py

```diff
@@ -79,3 +93,26 @@ def test_omegaconf_create(data: Any, benchmark: Any) -> None:
 )
 def test_omegaconf_merge(merge_function: Any, merge_data: Any, benchmark: Any) -> None:
     benchmark(merge_function, merge_data)
+
+
+@mark.parametrize(
+    "lst",
+    [
+        lazy_fixture("small_list"),
+        lazy_fixture("small_listconfig"),
+    ],
+)
+def test_list_in(lst: List[Any], benchmark: Any) -> None:
+    benchmark(lambda seq, val: val in seq, lst, 10)
+
+
+@mark.parametrize(
+    "lst",
+    [
+        lazy_fixture("small_list"),
+        lazy_fixture("small_listconfig"),
+    ],
+)
+def test_list_iter(lst: List[Any], benchmark: Any) -> None:
+    # Performance is really bad for list config iteration, not sure why.
```

Comment:
Why is that surprising? If you compare to a regular list, ListConfig is doing a lot more stuff to check if any interpolation needs to be resolved, while a regular list can instantly return the reference to the object.

Reply:
Because it's 500 times slower :).
I don't think it's justified. I tried to profile it a bit and do some blind optimizations but was not able to pinpoint the cause.

Reply:
Ok -- actually given the speed-up that you got here, I do agree that it should be possible to iterate faster (`x in lst` shouldn't be much faster than `for y in lst` -- at least when iterating through the whole list in both cases)